site/docs/spec.md

```diff
@@ -646,6 +646,18 @@ Each partition field in the fields list is stored as an object. See the table fo
 
 In some cases partition specs are stored using only the field list instead of the object format that includes the spec ID, like the deprecated `partition-spec` field in table metadata. The object format should be used unless otherwise noted in this spec.
 
+#### Partition Field ID handling
+
+A partition field id is an integer (starting at 1000) used to identify a partition field.
```

Comment:
The requirement isn't to start at 1000, it is that the IDs should not conflict with the IDs required for the other fields in data files. I think starting at 1000 is a good recommendation to avoid that.